### PR TITLE
Snobal 1d

### PR DESCRIPTION
**Pull request by dmragar**

### CODE REVIEW

**Comment by jomey** [comment]
Looks like we need to go over some basic setup of the repositories .gitignore and your local one on your computer.
In general, you should add things like `.ipynb_checkpoints` `.DS_Store` to your global git setup. Also once committed and then added to the git ignore file will artificially blow up the size of a repository.
Let's schedule a quick call next week and take care of that and then re-issue this PR. My bad for not thinking of this earlier.

**Comment by jomey** [comment]
Here an example for a `.gitignore` for my local setup: https://github.com/jomey/environment-setups/blob/master/local/gitignore

**Comment by dmragar** [comment]
Sounds good. I’m available Monday. 
 
cheers
Dillon

File: snobal_1d/station_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+            elif x > -0.5 and x < 0.0:
+                x = 0.75
+                return x
+                
+            elif x > 0.0 and x < 0.5:
+                x = 0.25
+                return x
+                           
+    def transform_density(x):
+        #if x != np.nan:
```
Comment (jomey) [inline snobal_1d/station_pipeline.py:173]: I would remove commented out code completely to make it easier to read.

File: snobal_1d/station_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    written by TH Painter, and provided by S Skiles.
+    
+    :param df: pandas df to pull temp col from
+    :param air_t_col: air temp col (in deg C)
+    :return: df with new cols "fraction" and "density"
+    """
+    df['fraction'] = df[air_t_col].copy()
+    df['density'] = df[air_t_col].copy()
+    
+    def transform_fraction(x):
```
Comment (jomey) [inline snobal_1d/station_pipeline.py:143]: Personally I avoid nesting functions within functions unless necessary. Usually they can be equally defined at same level and code is easier to read.

File: snobal_1d/station_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    Updated longwave estimation from Immerzeel et al. 2019. Currently implemented
+    with list comprehension is Irwin_pipeline.ipynb, because I had issues getting 
+    multiple df args to work w/ a lambda function.
+    
+    :param RH_col: col with RH
+    :param sw_net: col with net shortwave, in W/M^2
+    :param air_t_col_K: col with 
+    :return: list populated with estimated longwave.
+    """
+    SB_CONST = 5.67e-8
```
Comment (jomey) [inline snobal_1d/station_pipeline.py:71]: Very nice! Using all caps for constants is a good practice

Comment (jomey) [reply snobal_1d/station_pipeline.py:71]: Looks like this one is duplicate though. Sure you need to declare this again?

**Review by jomey — CHANGES_REQUESTED** [review]
Think you need to add the 'Icon' file to your `.gitignore` as well. It's an OS X specific file not needed under version control for others.

Suggest doing a quick cleanup of commented out code in `station_pipeline.py`.

File: snobal_1d/station_pipeline.py (hunk 1 above)
Comment (dmragar) [reply snobal_1d/station_pipeline.py]: Commented out code is removed. I've also added "Icon" to my `.gitignore`. What is the best way to remove it from this repo. Can I just delete it?

Comment (jomey) [reply snobal_1d/station_pipeline.py]: For cases like the icon, where the file size is 0 bytes, I would say yes. Just delete them and commit that change.

File: snobal_1d/station_pipeline.py (hunk 3 above)
Comment (dmragar) [reply snobal_1d/station_pipeline.py:70]: Not a duplicate; there are two different parameterizations. I will make it more clear in the docstring.

File: snobal_1d/station_pipeline.py (hunk 2 above)
Comment (dmragar) [reply snobal_1d/station_pipeline.py:142]: Good to know. I will avoid nesting in future.

File: snobal_1d/station_pipeline.py (hunk 3 above)
Comment (jomey) [reply snobal_1d/station_pipeline.py:70]: Help me out:
Line 15: `SB_CONST = 5.67e-8`
and you are re-defining it here. Where is the difference?

Comment (dmragar) [reply snobal_1d/station_pipeline.py]: Ah, the constant. Fixed!

**Review by jomey — CHANGES_REQUESTED** [review]
Can you please go one more time through the `Icon` files and remove them all?

Good resource to see if they are still in there is to go to the 'Files Changed' tab on the PR and look at the list of files included in this PR. There you will see that there are still some in there.

**Comment by dmragar** [comment]
> Can you please go one more time through the `Icon` files and remove them all?
> 
> Good resource to see if they are still in there is to go to the 'Files Changed' tab on the PR and look at the list of files included in this PR. There you will see that there are still some in there.

Good to know about the "Files Changed" tab! I removed the rest of the `Icon` files.

**Review by jomey — APPROVED** [review]

**Pull request merged**